simpcli/src/main.rs

```diff
@@ -104,6 +123,27 @@ fn main() -> Result<(), String> {
 
     // Execute command
     match command {
+        Command::Assemble => {
+            let prog = parse_file(&first_arg)?;
+
+            let roots = prog.roots();
+            let mut error = false;
+            for name in roots.keys() {
+                if **name != *"main" {
```

Comment:
bac6647aa3601812908c45642d48b884d38f11c3 `name.as_ref() != "main"` seems more canonical. I have an aversion against long sequences of `*` and `&`.